R-packages/covidcast/R/covidcast.R

```diff
@@ -515,32 +518,76 @@ summary.covidcast_meta = function(object, ...) {
   invisible(df)
 }
 
-##########
+# Find the maximum number of rows the API is expected to return per day of data
+max_geo_values <- function(data_source, signal, geo_type) {
+  meta_info <- covidcast_meta()
```

Comment:
`covidcast_signal()` already calls `covidcast_meta()`. Could it be more appropriate for `covidcast_signal()` to call `max_geo_values()` directly, providing it the necessary metadata, and then give `max_locations` as an argument to `covidcast_days`?

Otherwise each `covidcast_signal()` call would require two `covidcast_meta()` calls, which inflates API calls a fair bit.

Reply:
I combined `max_geo_values` with some filtering work that was already being done within `covidcast_signal` and renamed it to `specific_meta`. `max_geos` is now an optional parameter for `covidcast_days`, if not provided it assumes a large number of rows may be returned, so it queries one day at a time.